runtime/simulation/tapasco-start-sim.py

```diff
@@ -0,0 +1,169 @@
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+import errno
+import os
+from os import path
+import signal
+import sys
+from time import sleep
+import shutil
+import subprocess
+
+sim_dir = f"{os.environ['TAPASCO_WORK_DIR']}/simulation"
+
+class bcolors:
+    HEADER = '\033[95m'
+    OKBLUE = '\033[94m'
+    OKCYAN = '\033[96m'
+    OKGREEN = '\033[92m'
+    WARNING = '\033[93m'
+    FAIL = '\033[91m'
+    ENDC = '\033[0m'
+    BOLD = '\033[1m'
+    UNDERLINE = '\033[4m'
+
+    def _add_end(string):
+        return string + bcolors.ENDC
+
+    def h(string):
+        return bcolors._add_end(bcolors.HEADER + string)
+    def bl(string):
+        return bcolors._add_end(bcolors.OKBLUE + string)
+    def c(string):
+        return bcolors._add_end(bcolors.OKCYAN + string)
+    def g(string):
+        return bcolors._add_end(bcolors.OKGREEN + string)
+    def w(string):
+        return bcolors._add_end(bcolors.WARNING + string)
+    def f(string):
+        return bcolors._add_end(bcolors.FAIL + string)
+    def b(string):
+        return bcolors._add_end(bcolors.BOLD + string)
+    def u(string):
+        return bcolors._add_end(bcolors.UNDERLINE + string)
+
+
+def handler(sig, frame):
+    global make_vivado_prj
+    if make_vivado_prj is not None and make_vivado_prj.returncode is None:
+        grp = os.getpgid(make_vivado_prj.pid)
+        os.killpg(grp, signal.SIGTERM)
+
+    global make
+    if make is not None and make.returncode is None:
+        grp = os.getpgid(make.pid)
+        os.killpg(grp, signal.SIGTERM)
+
+    print('')
+    sys.exit(0)
+
+
+def target_vivado_prj(filename, verbose):
+    if not path.exists(filename):
+        print(bcolors.f(f'File not found: \'{filename}\''))
+        sys.exit(-1)
+
+    # copy mif file if exists
+    mif_dir = f'{sim_dir}/simulate_testbench.sim/sim_1/behav/questa'
+    mif_name = 'system_tapasco_status_base_0.mif'
+    if path.exists(f'{mif_dir}/{mif_name}'):
+        shutil.copyfile(f'{mif_dir}/{mif_name}', f'{sim_dir}/{mif_name}')
+
+    global make_vivado_prj
+
+    print('Creating Vivado files for simulation')
+
+    make_vivado_prj = subprocess.Popen([shutil.which('make'), '-C', sim_dir, f'SIM_IP={filename}', 'vivado_prj'],
+                                       preexec_fn=os.setsid,
+                                       stderr=subprocess.STDOUT if verbose == 2 else subprocess.PIPE,
```

Comment:
One minor thing that I just noticed while using the simulation in a broken environment: I would not want to have the user count how many verbosity levels he needs when debugging a simulation build. So I suggest the following change (again in the ``target_make`` function): 
```suggestion
                                       stderr=subprocess.STDOUT if verbose >= 2 else subprocess.PIPE,
```